### PR TITLE
TASK-94: Front-end File Size Validation

### LINKED ISSUE #322: TASK-94: Add Frontend Validation for Image Uploads
**Task Description**

As a developer, I want to add a warning/error stating that the image the user wants to upload is bigger than 10MB, and it must be smaller than that.

We are using image upload in:
* Create event
* All message pages
* Edit profile

**Acceptance Criteria**
- [ ] File Sizes are validated everywhere images can be uploaded
- [ ] Documentation present where necessary
- [ ] Unit and UI tests are implemented for every new component and function
- [ ] The application should still work with seeded data

### PR DESCRIPTION
**Pull request by r-saba**
Closes #322

Please ensure the acceptance criteria pass.

### CODE REVIEW

File: apps/mull-ui/src/utilities.ts [hunk 1]
```diff
@@ -52,6 +52,19 @@ export const avatarUrl = (user: Partial<User>) =>
     ? `${environment.backendUrl}/api/media/${user.avatar.id}`
     : `./assets/icons/icon-192x192.png`;
 
+export const validateFileSize = (file: File) => {
+  if (!file) {
+    return true;
+  }
+
+  const maxFileSize = 10000000;
```
Comment (cristian-aldea) [inline apps/mull-ui/src/utilities.ts]: Can you make a variable for this size? similar to the limits I created for the form fields for things such as the event title and description

File: apps/mull-ui/src/app/pages/create-event/create-event.tsx [hunk 2]
```diff
@@ -51,6 +51,8 @@ const CreateEventPage = ({ history }: CreateEventProps) => {
    * @param {ChangeEvent<HTMLInputElement>} event
    */
   const handleFileUpload = (event: ChangeEvent<HTMLInputElement>) => {
+    const { name: fileName, size: fileSize } = event.target.files[0];
+    console.log(fileSize);
```
Comment (osahtout) [inline apps/mull-ui/src/app/pages/create-event/create-event.tsx]: Should we remove this?

**Review by osahtout — APPROVED** [review]
Just the console log and whats suggested by C

File: apps/mull-ui/src/utilities.ts (hunk 1 above)
Comment (r-saba) [reply apps/mull-ui/src/utilities.ts]: Done

File: apps/mull-ui/src/app/pages/create-event/create-event.tsx (hunk 2 above)
Comment (r-saba) [reply apps/mull-ui/src/app/pages/create-event/create-event.tsx]: Done

File: apps/mull-ui/src/app/pages/create-event/create-event.tsx [hunk 3]
```diff
@@ -127,7 +127,9 @@ const CreateEventPage = ({ history }: CreateEventProps) => {
             return location && location.title && location.title !== '';
           }
         ),
-      imageFile: Yup.mixed().required('Image is required.'),
+      imageFile: Yup.mixed()
+        .required('Image is required.')
+        .test('big-file', 'File Size Too Large', validateFileSize),
```
Comment (cindyslittleplanet) [inline apps/mull-ui/src/app/pages/create-event/create-event.tsx]: Error message should not be in camel case to match other error messages. 
```suggestion
        .test('big-file', 'File size is too large.', validateFileSize),
```

**Review by cindyslittleplanet — APPROVED** [review]
LGTM! 🚀 Just one tiny nitpicking. 

**Acceptance Criteria**
- [x] File Sizes are validated everywhere images can be uploaded
* Create event
* All message pages
* Edit profile
- [x] Documentation present where necessary
- [x] Unit and UI tests are implemented for every new component and function
- [x] The application should still work with seeded data

File: apps/mull-ui/src/app/pages/create-event/create-event.tsx (hunk 3 above)
Comment (r-saba) [reply apps/mull-ui/src/app/pages/create-event/create-event.tsx]: Done

**Pull request merged**